### PR TITLE
Examplified module docs on ok.ex, culled ok_test.exs

### PR DESCRIPTION
**Pull request by bill-mybiz**
* I added a basic progression of examples using doctests in ok.ex.
  * I didn't cull as much from ok_test.exs as possible I don't think.
  * I removed repeated README basics from the module doc summary.
    * I linked to the README itself for more information.
* As of 1.5.0, hexdocs has nothing under "PAGES" or "MODULES"?
  * https://hexdocs.pm/ok/readme.html
  * 1.4.0 still has them.
    * https://hexdocs.pm/ok/1.4.0/OK.html#summary

### CODE REVIEW

File: lib/ok.ex [hunk 1]
```diff
@@ -72,65 +49,51 @@ defmodule OK do
...
-      iex> {:ok, 6} ~>> safe_div(0) ~>> double
-      {:error, :zero_division}
 
   It can be used in several ways.
   Pipe to a local call.
   This example is the same as calling `double(5)`
 
-      iex> {:ok, 5} ~>> double
-      {:ok, 10}
+    iex> {:ok, 5} ~>> double()
```
Comment (CrowdHailer) [inline lib/ok.ex]: These should be indented 4 spaces from the main block of text to ensure that they are formatted properly in module docs. If you are making lots of doc changes. which i LOVE you can try building them locally to see the output

Comment (bill-mybiz) [inline lib/ok.ex]: Ah, interesting. It didn't even occur to me Doh! I will definitely do that. :+1: 

Did you have a chance to checkout why the hexdocs lost the existing module docs?

**Comment by CrowdHailer** [comment]
I will look into hexdocs now, that is quite annoying

**Comment by CrowdHailer** [comment]
Fixed 1.5.0 but not the same problem for 1.5.1 !

**Comment by CrowdHailer** [comment]
@bill-mybiz turns out code snippets in headers is an issue. https://github.com/CrowdHailer/OK/commit/652ea6a58a36d206967f7ed60d660984f5c7f335

this has not always been the case

**Comment by bill-mybiz** [comment]
Good to know :thought_balloon:

File: lib/ok.ex (hunk 1 above)
Comment (CrowdHailer) [inline lib/ok.ex]: Yeah its just `mix docs` I think to build them locally. Took me a while to find that. one of my early libraries has a bunch of patch versions cz I was just pushing new docs. oops

Comment (CrowdHailer) [inline lib/ok.ex]: @bill-mybiz half waiting for you two fix the spaces here. and let me know docs work locally. ping me if your busy I can probably find the time soon.

**Comment by bill-mybiz** [comment]
Ok, I made those indentation adjustments and worked a little bit more on the overall flow. The `mix docs` looks good to me now so see what you think.

File: lib/ok.ex (hunk 1 above)
Comment (bill-mybiz) [inline lib/ok.ex]: @CrowdHailer Sorry about that. My brain just didn't fire an event handler on making those changes, and as I already mentioned gmail was basically filtering my github notifications...doh! I have just now updated the PR.

File: lib/ok.ex [hunk 2]
```diff
@@ -92,65 +74,52 @@ defmodule OK do
...
       iex> {:ok, 5} ~>> OKTest.double()
       {:ok, 10}
 
       iex> {:ok, 5} ~>> __MODULE__.double()
       {:ok, 10}
 
-  Pipe with extra arguments
-  This example is the same as calling `OK.safe_div(3, 4)`
+  Pipe with extra arguments.<br />
+  _(This is equivalent to calling `OK.safe_div(3, 4)`)_
```
Comment (CrowdHailer) [inline lib/ok.ex]: This is probably a mistake of mine from earlier but I don't see any 3's and 4's should this be equivalent to `OKTest.safe_div(6,2)`

File: lib/ok.ex [hunk 3]
```diff
@@ -92,65 +74,52 @@ defmodule OK do
...
+  _(This is equivalent to calling `double(5)`)_
 
-      iex> {:ok, 5} ~>> double
+      iex> {:ok, 5} ~>> double()
       {:ok, 10}
 
-  Pipe to a remote call.
-  This example is the same as calling `OKTest.double(3)`
+  Pipe to a remote call.<br />
+  _(This is equivalent to calling `OKTest.double(3)`)_
```
Comment (CrowdHailer) [inline lib/ok.ex]: `double(5)`?

**Comment by bill-mybiz** [comment]
:tada:

**Pull request merged**